### PR TITLE
A11y: Keep focus within the drawer when tabbing past the end

### PR DESCRIPTION
**Pull request by remybach**
Doing this as part of Ops Cops: https://trello.com/c/zsX8BtRD/908-accessibility-accreditation

> While navigating to items present under Hamburger Menu, when keyboard users complete the navigation under it, in next stroke after 'Sign In' link tab focus unnecessarily moves to the address bar and then in next two strokes focus goes to Close 'X' button.

Screengrab of me `shift`+`tab` / `tab`bing around the top/bottom of the drawer:

### CODE REVIEW

File: src/js/drawer.js [hunk 1]
```diff
@@ -68,19 +68,28 @@ function handleCloseEvents (scope, callback) {
...
+	};
+
 	function toggleCallback (state, e) {
 		if (state === 'close') {
 			toggleTabbing(drawerEl, false);
 
 			handleClose.removeEvents();
 
 			openingControl.focus();
+			drawerEl.removeEventListener('keydown', handleTab);
```
Comment (notlee) [inline src/js/drawer.js]: I'm not super familiar with `o-header`, but it looks like removing this event listener would belong in `handleClose.removeEvents();`?

File: src/js/drawer.js [hunk 2]
```diff
@@ -105,6 +114,7 @@ function addDrawerToggles (drawerEl) {
 					drawerEl.focus();
 				}
 			});
+			drawerEl.addEventListener('keydown', handleTab);
```
Comment (notlee) [inline src/js/drawer.js]: Similar to above, looks like `handleClose.addEvents` is added in a timeout to account for the transition. Does adding this event listener belong in that method?

Comment (remybach) [reply src/js/drawer.js]: Ah, yeah... When starting this I misunderstood how all the `handleClose` stuff worked, but I think you're right!

**Comment by notlee** [comment]
This looks good code wise and works well in Chrome. However I noticed it's not working well in Safari, as Safari [does not tab to anchors by default](http://www.weba11y.com/blog/2014/07/07/keyboard-navigation-in-mac-browsers/). I can't think of an approach to solve this off the top of my head though.

Kapture 2019-05-24 at 10 08 43

**Comment by i-like-robots** [comment]
This works in one direction but not the other, we should check if the shift key is pressed as well

**Pull request merged**